File: Temp-SPA/src/app/_models/application.ts
export interface Application {
    id: number;
    userId: number;
    teamId: number;
    moderatorId?: number;
    category: string;
    content: string;
}

export interface CreateApplication{
    userId: number;
    teamId: number;
    content: string;
    category: string;
}

export interface ModeratorListApplication {
    id: number;
    username: string;
    category: string;
    createdAt: Date;
    status: boolean;
}

export interface UserListApplication {
    id: number;
    category: string;
    createdAt: Date;
    status: boolean;
}

File: Temp.Domain/Models/Applications/Application.cs
namespace Temp.Domain.Models.Applications;

public class Application : BaseEntity
{
    public int Id { get; set; }

    public int UserId { get; set; }
    public Employee? User { get; set; }

    public int? ModeratorId { get; set; }
    public Employee? Moderator { get; set; }

    public int TeamId { get; set; }
    public Team? Team { get; set; }

    public string? Category { get; set; }

    public string? Content { get; set; }

    public bool Status { get; set; } = true;
    public DateTime StatusUpdatedAt { get; set; }
}